### DISCUSSION TITLE
Backtesting spread strategies

### DISCUSSION THREAD

**Discussion by Tullix95** [kernc/backtesting.py#899] [Q&A]
Hello everyone, 

I'm looking to build a strategy to trade spreads (FX, futures, ...), by first calculating the spread (for each security, open minus open, high minus high, etc. ). This may result in negative values (see screenshot below). 

![image](https://user-images.githubusercontent.com/105484046/216845983-2d93754e-4d54-4227-9cca-32324cdad1cd.png)

When I try to run the backtest, I get an error for negative equity (even after setting initial cash in the range of billions). 

![image](https://user-images.githubusercontent.com/105484046/216846113-20f7f093-28bd-4ebd-aaf9-ee5f1ebae7e1.png)

Is there a way to work this issue around using Backtesting.py? Or is there any alternative that allows for that?

Please let me know, 

Thank you

**Comment by kernc** [comment]
Have you tried computing the spread by subtracting bids from asks (`df_ask - df_bid`)? You shouldn't have been left with negative prices in that case.

Alternatively, you can try:
```py
ohlc_cols = ['Open', 'High', 'Low', 'Close']
df_spread[ohlc_cols] += df_spread[ohlc_cols].min() + eps

assert (df_spread[ohlc_cols] > 0).all().all()
